### PR TITLE
Fixes #791: SQS Queue names getting mangles

### LINKED ISSUE #791: SQS queue name gets mangled
Using Celery/Kombu in an environment with _future_ strings support you end up with mangled queue names in SQS.

The code that tries to translate invalid SQS queue characters causes the byte string to get converted to a string representation such as b'my_queue_name' and then translated that to 'b_my_queue_name'

I am finding this because we are using Apache Airflow in a Python 2.7 environment using SQS queues as the broker.

### PR DESCRIPTION
**Pull request by jkoehl**
Fixes #791 by slightly changing the order of when the queue name is translated to SQS valid name and when the unicode to str encoding happens. Basically doing the translate() on the original string instead of doing it after we have translated the string into an encoded unicode string.

### CODE REVIEW

File: kombu/async/http/curl.py [hunk 1]
```diff
@@ -218,9 +218,6 @@ def _setup_request(self, curl, request, buffer, headers, _pycurl=pycurl):
             if request.proxy_username:
                 setopt(_pycurl.PROXYUSERPWD, '{0}:{1}'.format(
                     request.proxy_username, request.proxy_password or ''))
-        else:
-            setopt(_pycurl.PROXY, '')
```
Comment (georgepsarakis) [inline kombu/async/http/curl.py]: Can you please explain this change? It seems unrelated to this PR.

File: kombu/transport/SQS.py [hunk 2]
```diff
@@ -154,10 +154,10 @@ def entity_name(self, name, table=CHARS_REPLACE_TABLE):
         """Format AMQP queue name into a legal SQS queue name."""
         if name.endswith('.fifo'):
             partial = name.rstrip('.fifo')
-            partial = text_t(safe_str(partial)).translate(table)
+            partial = text_t(safe_str(partial.translate(table)))
```
Comment (georgepsarakis) [inline kombu/transport/SQS.py]: Could you please a very simple test case to verify this change? You already have an example in https://github.com/celery/kombu/issues/791 , so you could use that.

File: kombu/async/http/curl.py (hunk 1 above)
Comment (jkoehl) [reply kombu/async/http/curl.py]: Sorry you are correct. That is a change for another PR. I branched from the wrong commit. I will fix that so this PR is only about this one issue.

**Comment by jkoehl** [comment]
I am going
